V1.37.1__add_address_book_service_endpoints migration failing

**Detailed Description**
Migration is failing when deployed to dev cluster. It seems like the assumption that `0.0.101` entries will have a port is not always true.

**Actual Behavior**
```
2021-05-10T12:40:00.250-0600 DEBUG main o.f.c.i.s.ParserSqlScript Found statement at line 5: -- normalize address_book_entry
-- if either node_id or node_account_id are unset, set them based on an applicable account id formatted memo value
-- only accountIds below 20 can be assumed to be offset with nodeId by 3
update address_book_entry
set node_account_id = split_part(memo, '.', 3)::int,
    node_id         = case
                          when split_part(memo, '.', 3)::int - 3 < 20 then split_part(memo, '.', 3)::int - 3
                          else node_id end
where node_account_id is null
  and memo ~ '^(\d{1,10}\.){2}\d{1,10}$'
2021-05-10T12:40:00.251-0600 DEBUG main o.f.c.i.s.ParserSqlScript Found statement at line 16: -- address_book_service_endpoint
create table if not exists address_book_service_endpoint
(
    consensus_timestamp bigint      not null,
    ip_address_v4       varchar(15) not null,
    node_id             bigint      not null,
    port                integer     not null
)
2021-05-10T12:40:00.255-0600 DEBUG main o.f.c.i.s.ParserSqlScript Found statement at line 24: comment on table address_book_service_endpoint is 'Network address book node service endpoints'
2021-05-10T12:40:00.256-0600 DEBUG main o.f.c.i.s.ParserSqlScript Found statement at line 26: -- add address_book_service_endpoint index
alter table address_book_service_endpoint
    add primary key (consensus_timestamp, node_id, ip_address_v4, port)
2021-05-10T12:40:00.257-0600 DEBUG main o.f.c.i.s.ParserSqlScript Found statement at line 30: -- migrate endpoints from address_book_entry to address_book_service_endpoint
insert into address_book_service_endpoint (consensus_timestamp, ip_address_v4, port, node_id)
select consensus_timestamp,
       ip,
       port,
       node_id
from address_book_entry
where node_id is not null
  and ip is not null
  and ip != ''
order by consensus_timestamp asc
2021-05-10T12:40:00.258-0600 DEBUG main o.f.c.i.s.ParserSqlScript Found statement at line 42: -- collapse duplicate node_id rows in address_book_entry
-- join table on self and remove duplicates with a lower id leaving a single instance of a memo-consensus_timestamp combo
delete
from address_book_entry a
    using address_book_entry b
where a.id < b.id
  and a.memo = b.memo
  and a.consensus_timestamp = b.consensus_timestamp
2021-05-10T12:40:00.259-0600 DEBUG main o.f.c.i.s.ParserSqlScript Found statement at line 51: -- Update address_book_entry
-- Add description, stake columns and primary key of consensus_timestamp and node_id
-- Update node_account_id and node_id to not be null
-- Drop id, ip and port columns
alter table if exists address_book_entry
    add column if not exists description varchar(100) null,
    add column if not exists stake       bigint       null,
    alter column node_account_id set not null,
    alter column node_id set not null,
    drop column if exists id,
    drop column if exists ip,
    drop column if exists port,
    add primary key (consensus_timestamp, node_id)
2021-05-10T12:40:00.259-0600 DEBUG main o.f.c.i.s.ParserSqlScript Found statement at line 65: -- update address_book node_count
with entry_map as (
    select consensus_timestamp, count(consensus_timestamp) from address_book_entry group by consensus_timestamp
)
update address_book
set node_count = count
from entry_map
where start_consensus_timestamp = entry_map.consensus_timestamp
2021-05-10T12:40:00.272-0600 INFO main o.f.c.i.c.DbMigrate Migrating schema "public" to version "1.37.1 - add address book service endpoints"
2021-05-10T12:40:00.276-0600 DEBUG main o.f.c.i.s.DefaultSqlScriptExecutor Executing SQL: -- normalize address_book_entry
-- if either node_id or node_account_id are unset, set them based on an applicable account id formatted memo value
-- only accountIds below 20 can be assumed to be offset with nodeId by 3
update address_book_entry
set node_account_id = split_part(memo, '.', 3)::int,
    node_id         = case
                          when split_part(memo, '.', 3)::int - 3 < 20 then split_part(memo, '.', 3)::int - 3
                          else node_id end
where node_account_id is null
  and memo ~ '^(\d{1,10}\.){2}\d{1,10}$'
2021-05-10T12:40:00.317-0600 DEBUG main o.f.c.i.s.DefaultSqlScriptExecutor Update Count: 4
2021-05-10T12:40:00.318-0600 DEBUG main o.f.c.i.s.DefaultSqlScriptExecutor Executing SQL: -- address_book_service_endpoint
create table if not exists address_book_service_endpoint
(
    consensus_timestamp bigint      not null,
    ip_address_v4       varchar(15) not null,
    node_id             bigint      not null,
    port                integer     not null
)
2021-05-10T12:40:00.371-0600 DEBUG main o.f.c.i.s.DefaultSqlScriptExecutor Update Count: 0
2021-05-10T12:40:00.372-0600 DEBUG main o.f.c.i.s.DefaultSqlScriptExecutor Executing SQL: comment on table address_book_service_endpoint is 'Network address book node service endpoints'
2021-05-10T12:40:00.379-0600 DEBUG main o.f.c.i.s.DefaultSqlScriptExecutor Update Count: 0
2021-05-10T12:40:00.379-0600 DEBUG main o.f.c.i.s.DefaultSqlScriptExecutor Executing SQL: -- add address_book_service_endpoint index
alter table address_book_service_endpoint
    add primary key (consensus_timestamp, node_id, ip_address_v4, port)
2021-05-10T12:40:00.396-0600 DEBUG main o.f.c.i.s.DefaultSqlScriptExecutor Update Count: 0
2021-05-10T12:40:00.398-0600 DEBUG main o.f.c.i.s.DefaultSqlScriptExecutor Executing SQL: -- migrate endpoints from address_book_entry to address_book_service_endpoint
insert into address_book_service_endpoint (consensus_timestamp, ip_address_v4, port, node_id)
select consensus_timestamp,
       ip,
       port,
       node_id
from address_book_entry
where node_id is not null
  and ip is not null
  and ip != ''
order by consensus_timestamp asc
2021-05-10T12:40:00.428-0600 DEBUG main o.f.c.i.j.TransactionalExecutionTemplate Rolling back transaction...
2021-05-10T12:40:00.429-0600 DEBUG main o.f.c.i.j.TransactionalExecutionTemplate Transaction rolled back
2021-05-10T12:40:00.430-0600 ERROR main o.f.c.i.c.DbMigrate Migration of schema "public" to version "1.37.1 - add address book service endpoints" failed! Changes successfully rolled back.
2021-05-10T12:40:00.438-0600 DEBUG main o.f.c.Flyway Memory usage: 57 of 183M
2021-05-10T12:40:00.438-0600 WARN main o.s.b.w.r.c.AnnotationConfigReactiveWebServerApplicationContext Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'flywayInitializer' defined in class path resource [org/springframework/boot/autoconfigure/flyway/FlywayAutoConfiguration$FlywayConfiguration.class]: Invocation of init method failed; nested exception is org.flywaydb.core.internal.command.DbMigrate$FlywayMigrateException:
Migration V1.37.1__add_address_book_service_endpoints.sql failed
----------------------------------------------------------------
SQL State  : 23502
Error Code : 0
Message    : ERROR: null value in column "port" violates not-null constraint
  Detail: Failing row contains (1600876594534358001, <ip>, 0, null).
Location   : db/migration/v1/V1.37.1__add_address_book_service_endpoints.sql (/app/resources/db/migration/v1/V1.37.1__add_address_book_service_endpoints.sql)
Line       : 30
Statement  : -- migrate endpoints from address_book_entry to address_book_service_endpoint
insert into address_book_service_endpoint (consensus_timestamp, ip_address_v4, port, node_id)
select consensus_timestamp,
       ip,
       port,
       node_id
from address_book_entry
where node_id is not null
  and ip is not null
  and ip != ''
order by consensus_timestamp asc

2021-05-10T12:40:00.586-0600 ERROR main o.s.b.SpringApplication Application run failed org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'flywayInitializer' defined in class path resource [org/springframework/boot/autoconfigure/flyway/FlywayAutoConfiguration$FlywayConfiguration.class]: Invocation of init method failed; nested exception is org.flywaydb.core.internal.command.DbMigrate$FlywayMigrateException: \\nMigration V1.37.1__add_address_book_service_endpoints.sql failed\\n----------------------------------------------------------------\\nSQL State  : 23502\\nError Code : 0\\nMessage    : ERROR: null value in column "port" violates not-null constraint\\n  Detail: Failing row contains (1600876594534358001, 34.94.106.61, 0, null).\\nLocation   : db/migration/v1/V1.37.1__add_address_book_service_endpoints.sql (/app/resources/db/migration/v1/V1.37.1__add_address_book_service_endpoints.sql)\\nLine       : 30\\nStatement  : -- migrate endpoints from address_book_entry to address_book_service_endpoint\\ninsert into address_book_service_endpoint (consensus_timestamp, ip_address_v4, port, node_id)\\nselect consensus_timestamp,\\n       ip,\\n       port,\\n       node_id\\nfrom address_book_entry\\nwhere node_id is not null\\n  and ip is not null\\n  and ip != ''\\norder by consensus_timestamp asc\\n\\n	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1786)\\n	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:602)\\n	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:524)\\n	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:335)\\n	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234)\\n	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:333)\\n	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208)\\n	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:322)\\n	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208)\\n	at org.springframework.context.support.AbstractApplicationContext.getBean(AbstractApplicationContext.java:1154)\\n	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:908)\\n	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:583)\\n	at org.springframework.boot.web.reactive.context.ReactiveWebServerApplicationContext.refresh(ReactiveWebServerApplicationContext.java:63)\\n	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:782)\\n	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:774)\\n	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:439)\\n	at org.springframework.boot.SpringApplication.run(SpringApplication.java:339)\\n	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1340)\\n	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1329)\\n	at com.hedera.mirror.importer.MirrorImporterApplication.main(MirrorImporterApplication.java:32)\\nCaused by: org.flywaydb.core.internal.command.DbMigrate$FlywayMigrateException: \\nMigration V1.37.1__add_address_book_service_endpoints.sql failed\\n----------------------------------------------------------------\\nSQL State  : 23502\\nError Code : 0\\nMessage    : ERROR: null value in column "port" violates not-null constraint\\n  Detail: Failing row contains (1600876594534358001, 34.94.106.61, 0, null).\\nLocation   : db/migration/v1/V1.37.1__add_address_book_service_endpoints.sql (/app/resources/db/migration/v1/V1.37.1__add_address_book_service_endpoints.sql)\\nLine       : 30\\nStatement  : -- migrate endpoints from address_book_entry to address_book_service_endpoint\\ninsert into address_book_service_endpoint (consensus_timestamp, ip_address_v4, port, node_id)\\nselect consensus_timestamp,\\n       ip,\\n       port,\\n       node_id\\nfrom address_book_entry\\nwhere node_id is not null\\n  and ip is not null\\n  and ip != ''\\norder by consensus_timestamp asc\\n\\n	at org.flywaydb.core.internal.command.DbMigrate.doMigrateGroup(DbMigrate.java:438)\\n	at org.flywaydb.core.internal.command.DbMigrate.access$200(DbMigrate.java:56)\\n	at org.flywaydb.core.internal.command.DbMigrate$3.call(DbMigrate.java:341)\\n	at org.flywaydb.core.internal.jdbc.TransactionalExecutionTemplate.execute(TransactionalExecutionTemplate.java:66)\\n	at org.flywaydb.core.internal.command.DbMigrate.applyMigrations(DbMigrate.java:338)\\n	at org.flywaydb.core.internal.command.DbMigrate.migrateGroup(DbMigrate.java:303)\\n	at org.flywaydb.core.internal.command.DbMigrate.access$100(DbMigrate.java:56)\\n	at org.flywaydb.core.internal.command.DbMigrate$2.call(DbMigrate.java:207)\\n	at org.flywaydb.core.internal.command.DbMigrate$2.call(DbMigrate.java:204)\\n	at org.flywaydb.core.internal.database.postgresql.PostgreSQLAdvisoryLockTemplate.execute(PostgreSQLAdvisoryLockTemplate.java:73)\\n	at org.flywaydb.core.internal.database.postgresql.PostgreSQLConnection.lock(PostgreSQLConnection.java:99)\\n	at org.flywaydb.core.internal.schemahistory.JdbcTableSchemaHistory.lock(JdbcTableSchemaHistory.java:139)\\n	at org.flywaydb.core.internal.command.DbMigrate.migrateAll(DbMigrate.java:204)\\n	at org.flywaydb.core.internal.command.DbMigrate.migrate(DbMigrate.java:165)\\n	at org.flywaydb.core.Flyway$1.execute(Flyway.java:215)\\n	at org.flywaydb.core.Flyway$1.execute(Flyway.java:164)\\n	at org.flywaydb.core.Flyway.execute(Flyway.java:538)\\n	at org.flywaydb.core.Flyway.migrate(Flyway.java:164)\\n	at org.springframework.boot.autoconfigure.flyway.FlywayMigrationInitializer.afterPropertiesSet(FlywayMigrationInitializer.java:66)\\n	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1845)\\n	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1782)\\n	... 19 more\\nCaused by: org.flywaydb.core.internal.sqlscript.FlywaySqlScriptException: \\nMigration V1.37.1__add_address_book_service_endpoints.sql failed\\n----------------------------------------------------------------\\nSQL State  : 23502\\nError Code : 0\\nMessage    : ERROR: null value in column "port" violates not-null constraint\\n  Detail: Failing row contains (1600876594534358001, 34.94.106.61, 0, null).\\nLocation   : db/migration/v1/V1.37.1__add_address_book_service_endpoints.sql (/app/resources/db/migration/v1/V1.37.1__add_address_book_service_endpoints.sql)\\nLine       : 30\\nStatement  : -- migrate endpoints from address_book_entry to address_book_service_endpoint\\ninsert into address_book_service_endpoint (consensus_timestamp, ip_address_v4, port, node_id)\\nselect consensus_timestamp,\\n       ip,\\n       port,\\n       node_id\\nfrom address_book_entry\\nwhere node_id is not null\\n  and ip is not null\\n  and ip != ''\\norder by consensus_timestamp asc\\n\\n	at org.flywaydb.core.internal.sqlscript.DefaultSqlScriptExecutor.handleException(DefaultSqlScriptExecutor.java:273)\\n	at org.flywaydb.core.internal.sqlscript.DefaultSqlScriptExecutor.executeStatement(DefaultSqlScriptExecutor.java:218)\\n	at org.flywaydb.core.internal.sqlscript.DefaultSqlScriptExecutor.execute(DefaultSqlScriptExecutor.java:126)\\n	at org.flywaydb.core.internal.resolver.sql.SqlMigrationExecutor.executeOnce(SqlMigrationExecutor.java:82)\\n	at org.flywaydb.core.internal.resolver.sql.SqlMigrationExecutor.access$000(SqlMigrationExecutor.java:32)\\n	at org.flywaydb.core.internal.resolver.sql.SqlMigrationExecutor$1.call(SqlMigrationExecutor.java:70)\\n	at org.flywaydb.core.internal.resolver.sql.SqlMigrationExecutor$1.call(SqlMigrationExecutor.java:67)\\n	at org.flywaydb.core.internal.database.DefaultExecutionStrategy.execute(DefaultExecutionStrategy.java:28)\\n	at org.flywaydb.core.internal.resolver.sql.SqlMigrationExecutor.execute(SqlMigrationExecutor.java:67)\\n	at org.flywaydb.core.internal.command.DbMigrate.doMigrateGroup(DbMigrate.java:434)\\n	... 39 more\\nCaused by: org.postgresql.util.PSQLException: ERROR: null value in column "port" violates not-null constraint\\n  Detail: Failing row contains (1600876594534358001, 34.94.106.61, 0, null).\\n	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2553)\\n	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2285)\\n	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:323)\\n	at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:481)\\n	at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:401)\\n	at org.postgresql.jdbc.PgStatement.executeWithFlags(PgStatement.java:322)\\n	at org.postgresql.jdbc.PgStatement.executeCachedSql(PgStatement.java:308)\\n	at org.postgresql.jdbc.PgStatement.executeWithFlags(PgStatement.java:284)\\n	at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:279)\\n	at org.flywaydb.core.internal.jdbc.JdbcTemplate.executeStatement(JdbcTemplate.java:241)\\n	at org.flywaydb.core.internal.sqlscript.ParsedSqlStatement.execute(ParsedSqlStatement.java:111)\\n	at org.flywaydb.core.internal.sqlscript.DefaultSqlScriptExecutor.executeStatement(DefaultSqlScriptExecutor.java:206)\\n	... 47 more
```

**Expected Behavior**
Migration to succeed

**Environment:**
- Java: AdoptOpenJDK 11.0.10
- OS: Kubernetes
- Version: v0.32.0-rc1

**Additional Context**

--- a/hedera-mirror-importer/src/main/resources/db/migration/v1/V1.37.1__add_address_book_service_endpoints.sql
+++ b/hedera-mirror-importer/src/main/resources/db/migration/v1/V1.37.1__add_address_book_service_endpoints.sql
@@ -19,7 +19,7 @@ create table if not exists address_book_service_endpoint
     consensus_timestamp bigint      not null,
     ip_address_v4       varchar(15) not null,
     node_id             bigint      not null,
-    port                integer     default 0 not null
+    port                integer     default -1 not null
 );
 comment on table address_book_service_endpoint is 'Network address book node service endpoints';
 

--- a/hedera-mirror-importer/src/main/resources/db/migration/v1/V1.37.1__add_address_book_service_endpoints.sql
+++ b/hedera-mirror-importer/src/main/resources/db/migration/v1/V1.37.1__add_address_book_service_endpoints.sql
@@ -31,7 +31,7 @@ alter table address_book_service_endpoint
 insert into address_book_service_endpoint (consensus_timestamp, ip_address_v4, port, node_id)
 select consensus_timestamp,
        ip,
-       case when port is null then 0 else port end,
+       case when port is null then -1 else port end,
        node_id
 from address_book_entry
 where node_id is not null

--- a/hedera-mirror-importer/src/main/resources/db/migration/v1/V1.37.1__add_address_book_service_endpoints.sql
+++ b/hedera-mirror-importer/src/main/resources/db/migration/v1/V1.37.1__add_address_book_service_endpoints.sql
@@ -19,7 +19,7 @@ create table if not exists address_book_service_endpoint
     consensus_timestamp bigint      not null,
     ip_address_v4       varchar(15) not null,
     node_id             bigint      not null,
-    port                integer     not null
+    port                integer     default 0 not null
 );
 comment on table address_book_service_endpoint is 'Network address book node service endpoints';
 
@@ -31,7 +31,7 @@ alter table address_book_service_endpoint
 insert into address_book_service_endpoint (consensus_timestamp, ip_address_v4, port, node_id)
 select consensus_timestamp,
        ip,
-       port,
+       case when port is null then -1 else port end,
        node_id
 from address_book_entry
 where node_id is not null

--- a/hedera-mirror-importer/src/main/resources/db/migration/v1/V1.37.1__add_address_book_service_endpoints.sql
+++ b/hedera-mirror-importer/src/main/resources/db/migration/v1/V1.37.1__add_address_book_service_endpoints.sql
@@ -31,7 +31,7 @@ alter table address_book_service_endpoint
 insert into address_book_service_endpoint (consensus_timestamp, ip_address_v4, port, node_id)
 select consensus_timestamp,
        ip,
-       case when port is null then -1 else port end,
+       case when port is null then 0 else port end,
        node_id
 from address_book_entry
 where node_id is not null

--- a/hedera-mirror-importer/src/main/resources/db/migration/v2/V2.0.0__time_scale_init.sql
+++ b/hedera-mirror-importer/src/main/resources/db/migration/v2/V2.0.0__time_scale_init.sql
@@ -56,7 +56,7 @@ create table if not exists address_book_service_endpoint
     consensus_timestamp bigint      not null,
     ip_address_v4       varchar(15) not null,
     node_id             bigint      not null,
-    port                integer     default 0 not null
+    port                integer     default -1 not null
 );
 comment on table address_book_service_endpoint is 'Network address book node service endpoints';
 

--- a/hedera-mirror-importer/src/main/resources/db/migration/v2/V2.0.0__time_scale_init.sql
+++ b/hedera-mirror-importer/src/main/resources/db/migration/v2/V2.0.0__time_scale_init.sql
@@ -56,7 +56,7 @@ create table if not exists address_book_service_endpoint
     consensus_timestamp bigint      not null,
     ip_address_v4       varchar(15) not null,
     node_id             bigint      not null,
-    port                integer     not null
+    port                integer     default 0 not null
 );
 comment on table address_book_service_endpoint is 'Network address book node service endpoints';
 

--- a/hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/migration/AddAddressBookServiceEndpointsMigrationTest.java
+++ b/hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/migration/AddAddressBookServiceEndpointsMigrationTest.java
@@ -305,6 +305,56 @@ class AddAddressBookServiceEndpointsMigrationTest extends IntegrationTest {
                 .isEmpty();
     }
 
+    @Test
+    void verifyAddressBookWithValidIpAndInvalidPortMigration() throws IOException {
+        long consensusTimestamp = 1;
+        AddressBookEntry.AddressBookEntryBuilder builder = AddressBookEntry.builder()
+                .nodeCertHash("nodeCertHash".getBytes())
+                .publicKey("rsa+public/key");
+
+        List<Long> nodeIds = List.of(0L, 1L, 2L, 3L);
+        insertAddressBook(AddressBookServiceImpl.ADDRESS_BOOK_102_ENTITY_ID, consensusTimestamp, nodeIds.size());
+        insertAddressBookEntry(
+                builder.memo(baseAccountId + (nodeIds.get(0) + nodeAccountOffset))
+                        .id(new AddressBookEntry.Id(consensusTimestamp, nodeIds.get(0)))
+                        .build(), "127.0.0.1", null);
+        insertAddressBookEntry(builder.memo(baseAccountId + (nodeIds.get(1) + nodeAccountOffset))
+                .id(new AddressBookEntry.Id(consensusTimestamp, nodeIds.get(1)))
+                .build(), "127.0.0.2", null);
+        insertAddressBookEntry(builder.memo(baseAccountId + (nodeIds.get(2) + nodeAccountOffset))
+                .id(new AddressBookEntry.Id(consensusTimestamp, nodeIds.get(2)))
+                .build(), "127.0.0.3", null);
+        insertAddressBookEntry(builder.memo(baseAccountId + (nodeIds.get(3) + nodeAccountOffset))
+                .id(new AddressBookEntry.Id(consensusTimestamp, nodeIds.get(3)))
+                .build(), "127.0.0.4", null);
+
+        runMigration();
+
+        IterableAssert<AddressBookEntry> listAssert =
+                assertThat(addressBookEntryRepository
+                        .findAll())
+                        .isNotEmpty()
+                        .hasSize(nodeIds.size());
+
+        listAssert.extracting(AddressBookEntry::getId).extracting(AddressBookEntry.Id::getNodeId)
+                .containsExactlyInAnyOrderElementsOf(nodeIds);
+        listAssert.extracting(AddressBookEntry::getNodeAccountId)
+                .containsExactlyInAnyOrder(
+                        EntityId.of(baseAccountId + (nodeIds.get(0) + nodeAccountOffset), EntityTypeEnum.ACCOUNT),
+                        EntityId.of(baseAccountId + (nodeIds.get(1) + nodeAccountOffset), EntityTypeEnum.ACCOUNT),
+                        EntityId.of(baseAccountId + (nodeIds.get(2) + nodeAccountOffset), EntityTypeEnum.ACCOUNT),
+                        EntityId.of(baseAccountId + (nodeIds.get(3) + nodeAccountOffset), EntityTypeEnum.ACCOUNT));
+
+        assertThat(addressBookServiceEndpointRepository
+                .findAll())
+                .isNotEmpty()
+                .hasSize(nodeIds.size())
+                .extracting(AddressBookServiceEndpoint::getId)
+                .extracting(AddressBookServiceEndpoint.Id::getNodeId)
+                .extracting(EntityId::getId)
+                .containsExactlyInAnyOrder(0L, 1L, 2L, 3L);
+    }
+
     private List<AddressBookEntry> getAndSaveAddressBookEntries(boolean deprecatedIp, long consensusTimestamp,
                                                                 int nodeCount, int endPointCount) {
         List<AddressBookEntry> addressBookEntries = new ArrayList<>();
@@ -388,7 +438,7 @@ class AddAddressBookServiceEndpointsMigrationTest extends IntegrationTest {
      * @param ip               service endpoint ip
      * @param port             service endpoint  port
      */
-    private void insertAddressBookEntry(AddressBookEntry addressBookEntry, String ip, int port) {
+    private void insertAddressBookEntry(AddressBookEntry addressBookEntry, String ip, Integer port) {
         Long nodeAccountId = EntityId.isEmpty(addressBookEntry.getNodeAccountId()) ? null :
                 addressBookEntry.getNodeAccountId().getId();
         jdbcOperations

--- a/hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/migration/AddAddressBookServiceEndpointsMigrationTest.java
+++ b/hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/migration/AddAddressBookServiceEndpointsMigrationTest.java
@@ -345,14 +345,20 @@ class AddAddressBookServiceEndpointsMigrationTest extends IntegrationTest {
                         EntityId.of(baseAccountId + (nodeIds.get(2) + nodeAccountOffset), EntityTypeEnum.ACCOUNT),
                         EntityId.of(baseAccountId + (nodeIds.get(3) + nodeAccountOffset), EntityTypeEnum.ACCOUNT));
 
-        assertThat(addressBookServiceEndpointRepository
-                .findAll())
-                .isNotEmpty()
-                .hasSize(nodeIds.size())
-                .extracting(AddressBookServiceEndpoint::getId)
+        IterableAssert<AddressBookServiceEndpoint> serviceListAssert =
+                assertThat(addressBookServiceEndpointRepository
+                        .findAll())
+                        .isNotEmpty()
+                        .hasSize(nodeIds.size());
+
+        serviceListAssert.extracting(AddressBookServiceEndpoint::getId)
                 .extracting(AddressBookServiceEndpoint.Id::getNodeId)
                 .extracting(EntityId::getId)
                 .containsExactlyInAnyOrder(0L, 1L, 2L, 3L);
+
+        serviceListAssert.extracting(AddressBookServiceEndpoint::getId)
+                .extracting(AddressBookServiceEndpoint.Id::getPort)
+                .containsExactlyInAnyOrder(0, 0, 0, 0);
     }
 
     private List<AddressBookEntry> getAndSaveAddressBookEntries(boolean deprecatedIp, long consensusTimestamp,

--- a/hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/migration/AddAddressBookServiceEndpointsMigrationTest.java
+++ b/hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/migration/AddAddressBookServiceEndpointsMigrationTest.java
@@ -320,13 +320,13 @@ class AddAddressBookServiceEndpointsMigrationTest extends IntegrationTest {
                         .build(), "127.0.0.1", null);
         insertAddressBookEntry(builder.memo(baseAccountId + (nodeIds.get(1) + nodeAccountOffset))
                 .id(new AddressBookEntry.Id(consensusTimestamp, nodeIds.get(1)))
-                .build(), "127.0.0.2", null);
+                .build(), "127.0.0.2", 0);
         insertAddressBookEntry(builder.memo(baseAccountId + (nodeIds.get(2) + nodeAccountOffset))
                 .id(new AddressBookEntry.Id(consensusTimestamp, nodeIds.get(2)))
                 .build(), "127.0.0.3", null);
         insertAddressBookEntry(builder.memo(baseAccountId + (nodeIds.get(3) + nodeAccountOffset))
                 .id(new AddressBookEntry.Id(consensusTimestamp, nodeIds.get(3)))
-                .build(), "127.0.0.4", null);
+                .build(), "127.0.0.4", 50211);
 
         runMigration();
 
@@ -358,7 +358,7 @@ class AddAddressBookServiceEndpointsMigrationTest extends IntegrationTest {
 
         serviceListAssert.extracting(AddressBookServiceEndpoint::getId)
                 .extracting(AddressBookServiceEndpoint.Id::getPort)
-                .containsExactlyInAnyOrder(0, 0, 0, 0);
+                .containsExactlyInAnyOrder(-1, 0, -1, 50211);
     }
 
     private List<AddressBookEntry> getAndSaveAddressBookEntries(boolean deprecatedIp, long consensusTimestamp,

